components/director/internal/authenticator/middleware.go

```diff
@@ -93,6 +100,21 @@ func (a *Authenticator) Handler() func(next http.Handler) http.Handler {
 				return
 			}
 
+			if mdc := log.MdcFromContext(ctx); nil != mdc {
+				if tokenClaims.ConsumerType != "" {
```

Comment:
To me, this part looks a bit ugly with those 4 ifs. I'd extract it in a separate function or create a "SetIfNotEmpty" method to at least get rid of the ifs.

Reply:
fixed